synapse/lib/types.py

```diff
@@ -713,7 +723,13 @@ def norm(self, valu):
             raise s_exc.BadTypeValu(name=self.name, valu=valu,
                                     mesg=str(e)) from None
 
-        huge = s_common.hugenum(valu)
+        # behave modulo like int/float
+        if self.modulo is not None:
+            huge = huge % self.modulo
```

Comment:
Need to add a `s_common.hugemod` that uses `hugectx.divmod` to avoid precision errors

Reply:
Actually, we do this before checking hugemin/hugemax so we may need to catch exceptions anyway since they could shove a huge value in and get a `decimal.InvalidOperation: [<class 'decimal.DivisionImpossible'>]`